SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditorLayoutDetails.cpp

```diff
@@ -101,8 +101,21 @@ FReply FSpatialGDKEditorLayoutDetails::GenerateDevAuthToken()
 		return FReply::Unhandled();
 	}
 
-	TSharedPtr<FJsonObject> JsonDataObject = JsonRootObject->GetObjectField("json_data");
-	FString TokenSecret = JsonDataObject->GetStringField("token_secret");
+	const TSharedPtr<FJsonObject> *JsonDataObject;
```

Comment:
Style nits - put the * on the type instead of on the variable. Also a pointer to a shared pointer is a really weird object to have kicking around. What is the ownership of this meant to be doing? Is the thing this is passed into meant to give you ownership over the object? If so then this should be a shared pointer and we take an address when passing it into the function below.

Reply:
If not then why is this a shared pointer (which is generally a smell at the best of times). That's a really weird API.

Reply:
Ok. I've thought about it and the fact that it's a pointer to const means it's unlikely the caller is meant to have shared ownership (you can still mutate the underlying pointer in a const shared pointer but can't set it to something else).
So a pointer to shared pointer might be correct. This raises the question of why the API tells you that stored in a shared pointer but that's just something we have to live with.
In which case I think this might need to be kept like this but moving the * to the left and adding a comment saying that this is just a quirk of the json API.